indico/modules/rb_new/controllers/backend/bookings.py

```diff
@@ -276,9 +276,40 @@ def _check_access(self):
         if not self.booking.can_edit(session.user):
             raise Forbidden
 
+    def _should_split(self, new_start_dt, new_end_dt):
+        today = date.today()
+        is_ongoing_booking = new_start_dt.date() < today
+        old_start_time, old_end_time = self.booking.start_dt.time(), self.booking.end_dt.time()
+        times_changed = new_start_dt.time() != old_start_time or new_end_dt.time() != old_end_time
+        return is_ongoing_booking and times_changed
+
+    def split_bookings(self, booking_data):
+        occurrences = self.booking.occurrences
+        room = self.booking.room
+        is_prebooking = not self.booking.is_accepted
+        occurrences_to_cancel = filter(lambda r: r.start_dt.date() >= date.today(), occurrences)
+        new_start_dt = datetime.combine(occurrences_to_cancel[0].start_dt.date(), booking_data['start_dt'].time())
```

Comment:
that's not reliable, `booking.occurrences` is does not have any particular order. btw, call `.all()` when getting `occurrences` above to avoid running the query multiple times.

Reply:
shouldn't the start date of the new reservation be the one of its first occurrence?